src/eei.cpp

```diff
@@ -550,6 +602,14 @@ namespace HeraVM {
       if (call_result.release)
         call_result.release(&call_result);
 
+      if (!isZeroUint256(call_message.value))
+        takeGas(GasSchedule::valuetransfer);
+
+      if (import->base == Name("call") && !context->fn_table->account_exists(context, &call_message.address))
+        takeGas(GasSchedule::callnewacct);
+
+      takeGas(GasSchedule::call);
```

Comment:
1. This should go first, before other checks.
2. Also, the gas should be taken _before_ the actual call.
3. You also have to take gas of the gas argument, and put back the amount left after the call.